mondrian/src/main/java/mondrian/rolap/agg/SegmentCacheManager.java

```diff
@@ -1606,13 +1606,6 @@ public SegmentCacheIndex getIndex(RolapStar star) {
         private SegmentCacheIndex getIndex(
             SegmentHeader header)
         {
-            // First we check the indexes that already exist.
```

Comment:
Because `Checksum` was replaced by a `SchemaKey`, and we can retreive `SchemaKey` only from `star.getSchema().getKey()`.
If I will replace `header.schemaChecksum` with the `getStar(header).getSchema().getKey()` we will get the duplication of `getIndex(star)`